honeybee_energy/ventcool/opening.py

```diff
@@ -34,25 +34,59 @@ class VentilationOpening(object):
             cross ventilation will be induced. If False, the assumption is that
             the operable area is primarily on one side of the Room and there is
             no wind-driven ventilation. (Default: False)
+        air_mass_flow_coefficient_closed: A number in kg/s-m, at 1 Pa per meter of crack
+            length, used to calculate the mass flow rate when the opening is closed;
+            required to run an AirflowNetwork simulation. Some common values for this
+            coefficient from the DesignBuilder Cracks template include the following:
+
+                * 0.00001 - Tight, low-leakage closed external window
+                * 0.003 - Very poor, high-leakage closed external window
+
+            If this value is None and one of the AirflowNetwork options are
+            selected in the VentilationSimulationControl object, a default value will be
+            calculated to produce air flow leakages equivalent to the zone flow rate
+            defined by the corresponding Infiltration object.
```

Comment:
No, I don't think that's not what is going to happen here with a `None` input. Where do you propose this auto-calculation code is going to be? And how do you even know that a Face has a parent Room with an infiltration object? Just set it to 0 by default and you'll add a method on the RoomEnergyProperties that users can explicitly call to compute the flow coefficient this from the Room's infiltration object if it exists.

Or, maybe a wiser choice is always to leave this at 0 for such cases since you can probably account for the whole Room infiltration with just the parent Face'c crack.

Also, you should state the default value here in the docstring.

Reply:
@chriswmackey 

**Re: setting flow_coefficient to 0 as default**
Technically the flow_coefficient_closed value has to be greater than 0 (there's a float_positive check to ensure this). If the goal is to get rid of None as a default, I think there are three alternatives:

1. Allow a zero value. This isn't ideal because it could cause confusion if we allow 0 here (even if we are planning on overwriting it), and EnergyPlus doesn't.  The flow_exponent_exponent can't be set to zero either since that'll make the flow rate equal to 1 and there's a minimum value of 0.5. 

2. Set the flow_coefficient to some assumed default based on Face type and boundary condition. The flow_coefficient for openings has a range from 0.00001 to 0.04 (from DesignBuilder) based on if it's an interior/exterior window/door, so we'll have to apply this default based on the host aperture properties. I can include a private method in vent_opening that extracts this value from the host. Then this default (which will have to be based on the DB crack template references) can be overwritten by the auto-calc method in the RoomEnergyProperties if the user wants it to simulate the zone infiltration flow rate.

3. Add a very small value as a default like 0.00001. So by default we represent a very tight envelope. The problem with this is that it doesn't correlate with the default envelope properties for different Face types/boundary conditions. So it's a unneccessarily inaccurate default. 

I'm like option number 2, but am open to another options.

**Re: Accounting the whole Room infiltration with just the parent Face crack.**  
Right now I'm translating the zone infiltration for leakage through opening edges AND surface area cracks:
https://github.com/ladybug-tools/honeybee-energy/pull/304/files#diff-cd4564ea076fa96e5cb3a05f5a09915eR248-R278. Since the blower door tests used to derive the zone infiltration rates were done one buildings with a mix of surface cracks and opening edges, this will give us relatively more accurate surface/edge opening parameters. In terms of the resulting infiltration load there shouldn't be any difference between these approaches, but it could be misleading if someone were looking at the Face crack coefficients.  

**Re: RoomEnergyProperties auto-calculation from infiltration**
As for the auto-calculation, right now it's just a function in the afn.py module (you can see it above if you scroll up), but I don't see any problem moving the code to RoomEnergyProperties.

Reply:
I am definitely leaning towards option 1 as the other two seem to be making some dangerous implicit assumptions for the user without their awareness.  And having a default of no infiltration is consistent with how everything else in honeybee works. For example, if someone initializes an empty `ProgramType` object, the object has no infiltration until the user specifies it. The same thing should be true of the AFN and the user should have to explicitly specify the flow coefficient or run some method to autocalculate it based on other properties.

Whatever the case, a flow coefficient of 0 should definitely be permissible since there has to be some way for the user to switch off the connection of a certain object to the AFN if they want to.  And bear in mind that this default of 0 isn't something that gets exposed on the Grasshopper components (we'll always run the autocalculation for the user if they request to use the AFN). It's just that, on the Python SDK layer we should be as explicit as possible.

Intuitively, it feels like the Room auto-calculation should be method on `RoomEnergyProperties` since the method doesn't get applied to anything else.  I'll take a look over it now.

Reply:
Okay, and the inability to translate the optional None to C# argument you make below is convincing. I'll switch all these to 0, and document our assumptions.